scripts/eosio_build_centos.sh

```diff
@@ -192,112 +194,200 @@ printf "\\n"
+if $PIN_COMPILER; then
+   printf "Checking Clang 8 support...\\n"
+   if [ ! -d $CLANG8_ROOT ] || [ $FORCE_BUILD ]; then
+      printf "Installing Clang 8...\\n"
+      cd ${TMP_LOCATION} \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/llvm.git clang8 && cd clang8 \
+      && git checkout $PINNED_COMPILER_LLVM_COMMIT \
+      && cd tools \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/lld.git \
+      && cd lld && git checkout $PINNED_COMPILER_LLD_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/polly.git \
+      && cd polly && git checkout $PINNED_COMPILER_POLLY_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/clang.git clang && cd clang/tools \
+      && git checkout $PINNED_COMPILER_CLANG_VERSION \
+      && mkdir extra && cd extra \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/clang-tools-extra.git \
+      && cd clang-tools-extra && git checkout $PINNED_COMPILER_CLANG_TOOLS_EXTRA_COMMIT && cd .. \
+      && cd ../../../../projects \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/libcxx.git \
+      && cd libcxx && git checkout $PINNED_COMPILER_LIBCXX_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/libcxxabi.git \
+      && cd libcxxabi && git checkout $PINNED_COMPILER_LIBCXXABI_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/libunwind.git \
+      && cd libunwind && git checkout $PINNED_COMPILER_LIBUNWIND_COMMIT && cd ../ \
+      && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/compiler-rt.git \
+      && cd compiler-rt && git checkout $PINNED_COMPILER_COMPILER_RT_COMMIT && cd ../ \
+      && cd ${TMP_LOCATION}/clang8 \
+      && mkdir build && cd build \
+      && $CMAKE -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX="${CLANG8_ROOT}" -DLLVM_BUILD_EXTERNAL_COMPILER_RT=ON -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_ENABLE_LIBCXX=ON -DLLVM_ENABLE_RTTI=ON -DLLVM_INCLUDE_DOCS=OFF -DLLVM_OPTIMIZED_TABLEGEN=ON -DLLVM_TARGETS_TO_BUILD=host -DCMAKE_BUILD_TYPE=Release .. \
+      && make -j"${JOBS}" \
+      && make install \
+      && rm -rf ${TMP_LOCATION}/clang8 \
+      && cd ../.. \
+      || exit 1
+
+      printf " - Clang 8 successfully installed @ ${CLANG8_ROOT}\\n"
+   else
+      printf " - Clang 8 found @ ${CLANG8_ROOT}.\\n"
+   fi
+   if [ $? -ne 0 ]; then exit -1; fi
+
+   printf "\\n"
+
+   printf "Checking LLVM 4 installation...\\n"
+   if [ ! -d $OPT_LOCATION/llvm4 ] || [ $FORCE_BUILD ]; then
+      printf "Installing LLVM 4...\\n"
+      cd $SRC_LOCATION \
+      && git clone https://github.com/llvm-mirror/llvm --single-branch --branch $LLVM_VERSION && cd llvm \
```

Comment:
`FORCE_BUILD` is documented as "Force rebuilding of dependencies" but there are many places throughout the scripts where that doesn't actually work. This is one case, where when the clone is run a second time git errors out with `fatal: destination path 'llvm' already exists and is not an empty directory.` There are also places where tar just untars over an existing directory. It seems like for `FORCE_BUILD` to do what it says on the tin all these places needed to be further cleaned up to remove the source dir before doing the clone/untar/whatevs